invenio_records_rest/utils.py

```diff
@@ -34,6 +34,18 @@
 from invenio_records.api import Record
 from werkzeug.routing import BaseConverter, BuildError
 from werkzeug.utils import cached_property, import_string
+from functools import wraps
+from jsonschema.exceptions import ValidationError
+
+
+def catch_record_validation_error(func):
```

Comment:
Probably better suited as an error handler on the blueprint: http://flask.pocoo.org/docs/0.10/api/#flask.Blueprint.errorhandler